python/lsst/ap/association/packageAlerts.py

```diff
@@ -186,24 +190,49 @@ def createDiaSourceBBox(self, bboxSize):
             bbox = geom.Extent2I(bboxSize, bboxSize)
         return bbox
 
-    def createCcdDataCutout(self, cutout, skyCenter, photoCalib):
-        """Convert a cutout into a calibrate CCDData image.
+    def createCcdDataCutout(self, image, skyCenter, bbox, photoCalib, srcId):
+        """Grab an image as a cuttout and return a calibrated CCDData image.
 
         Parameters
         ----------
-        cutout : `lsst.afw.image.ExposureF`
-            Cutout to convert.
+        image : `lsst.afw.image.ExposureF`
+            Image to pull cutout from.
         skyCenter : `lsst.geom.SpherePoint`
             Center point of DiaSource on the sky.
+        bbox : `lsst.geom.Extent2I`
+            Bounding box to cutout from the image.
         photoCalib : `lsst.afw.image.PhotoCalib`
             Calibrate object of the image the cutout is cut from.
+        srcId : `int`
+            Unique id of DiaSource. Used for when an error occurs extracting
+            a cutout.
 
         Returns
         -------
-        ccdData : `astropy.nddata.CCDData`
+        ccdData : `astropy.nddata.CCDData` or `None`
             CCDData object storing the calibrate information from the input
             difference or template image.
         """
+        # Catch errors in retrieving the cutout.
+        try:
+            cutout = image.getCutout(skyCenter, bbox)
+        except InvalidParameterError:
+            point = image.getWcs().skyToPixel(skyCenter)
+            imBBox = image.getBBox()
+            self.log.warn(
+                "Failed to retrieve cutout from image for DiaSource with "
+                "id=%i. InvalidParameterError thrown during cutout creation. "
+                "Returning `None` for cuttout."
+                % srcId)
+            if not geom.Box2D(image.getBBox()).contains(point):
+                self.log.warn(
+                    "DiaSource centroid lies at pixel (%.2f, %.2f) "
+                    "which is outside the Exposure with bounding box "
+                    "((%i, %i), (%i, %i))." %
+                    (point.x, point.y,
+                     imBBox.minX, imBBox.maxX, imBBox.minY, imBBox.maxY))
```

Comment:
I'm a little worried about this swallowing up other as-yet-unknown errors. Would it make sense to only return `None` for this case, and re-raise the error if it was a different problem?

Reply:
So I looked over the code for cutouts in C++. The two errors this block can throw are both InvalidParameterError one for centroid outside of exposure bbox and the extent of the cutout bbox is <=0. There may be others further up stream but I haven't investigated that completely. I guess the best I can do here is restructure it to catch the error for outside centroid (by testing it) and then do a raise on anything else. Sound good?